modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RatedDocument.java

```diff
@@ -19,36 +19,69 @@
 
 package org.elasticsearch.index.rankeval;
 
+import org.elasticsearch.action.support.ToXContentToBytes;
+import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.ParsingException;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.common.xcontent.XContentParser.Token;
 
 import java.io.IOException;
+import java.util.Objects;
 
 /**
  * A document ID and its rating for the query QA use case.
  * */
-public class RatedDocument implements Writeable {
+public class RatedDocument extends ToXContentToBytes implements Writeable {
 
-    // TODO augment with index name and type name
-    private final String docId;
-    private final int rating;
+    public static final ParseField RATING_FIELD = new ParseField("rating");
+    public static final ParseField KEY_FIELD = new ParseField("key");
 
-    public RatedDocument(String docId, int rating) {
-        this.docId = docId;
+    private static final ObjectParser<RatedDocument, RankEvalContext> PARSER = new ObjectParser<>("ratings", RatedDocument::new);
+
+    static {
+        PARSER.declareObject(RatedDocument::setKey, (p, c) -> {
+            try {
+                return RatedDocumentKey.fromXContent(p, c);
+            } catch (IOException ex) {
+                throw new ParsingException(p.getTokenLocation(), "error parsing rank request", ex);
+            }
+        } , KEY_FIELD);
+        PARSER.declareInt(RatedDocument::setRating, RATING_FIELD);
+    }
+
+    private RatedDocumentKey key;
+    private int rating;
+
+    RatedDocument() {}
```

Comment:
Can we make the key and rating mandatory ctor args if this doesn't create too much overhead? The parser needs to be changed to ConstructingObjectParser then.